internal/cache/cache.go

```diff
@@ -18,15 +18,36 @@ package cache
 
 import "sync"
 
-type Cache[T any] struct {
+type Cache[T any] interface {
+	Get(key string) (T, bool)
+	Set(key string, entries T)
+	Delete(key string)
+}
+
+type DefaultCache[T any] struct {
```

Comment:
nit: maybe move the struct declaration down to line 46 to group with its methods?